shared/middleware/net_sim_api.rb

```diff
@@ -203,25 +209,19 @@ def owns_shard?(shard_id)
     begin
       body = JSON.parse(request.body.read)
     rescue JSON::ParserError
-      json_bad_request
+      json_bad_request(INSERT_ERRORS[:malformed])
     end
 
     # Determine whether or not we are performing a multi-insert and
     # normalize our request body into an array of values
-    if body.is_a?(Array)
-      multi_insert = true
-      values = body
-    elsif body.is_a?(Hash)
-      multi_insert = false
-      values = [body]
-    else
-      json_bad_request
-    end
+    multi_insert = body.is_a?(Array)
+    values = multi_insert ? body : [body]
 
-    # Validation
-    #   currently, only messages are validated
-    all_values_valid = values.all? { |value| value_valid?(shard_id, table_name, value) }
-    json_bad_request unless all_values_valid
+    validation_errors = validate_all(shard_id, table_name, values)
```

Comment:
Hmm... I wasn't going to tackle that in this PR, though it should happen at some point.  I can't just run the validation function inside update because some of the checks (`routerNumber` unique constraint in particular) wouldn't work as-is.